### PR TITLE
New package: hyperrogue-11.3f

### PR DESCRIPTION
**Pull request by xelxebar**

### CODE REVIEW

**Comment by xelxebar** [comment]
For some reason, `/usr/include/SDL/SDL_gfxPrimitives.h` from `SDL_gfx-devel` has
    #include <SDL.h>
whereas `SDL-devel` installs that header to `/usr/include/SDL/SDL.h`. That is why `pre_build` has that symlink hack.

**Comment by xelxebar** [comment]
Ugh. Crossbuilds are broken. Looks like upstream is generating some source from one of the output binaries, thus said binary needs to run on the host instead of target.

File: srcpkgs/hyperrogue/template [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+pkgname=hyperrogue
+version=11.3f
+revision=1
+archs="x86_64*"
+build_style=gnu-configure
+hostmakedepends="autoconf automake"
+makedepends="glew-devel libpng-devel SDL-devel SDL_gfx-devel SDL_mixer-devel SDL_ttf-devel zlib-devel"
+short_desc="SDL rogue-like in a non-euclidean world"
+maintainer="B. Wilson <x@wilsonb.com>"
+license="GPL-3.0-or-later"
```
Comment (travankor) [inline srcpkgs/hyperrogue/template]: gpl 2

File: srcpkgs/hyperrogue/template [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Template file for 'hyperrogue'
+pkgname=hyperrogue
+version=11.3f
+revision=1
+archs="x86_64*"
```
Comment (travankor) [inline srcpkgs/hyperrogue/template]: this shouldn't be set unless you are 100% sure this only works exclusively on that arch, which I don't think is the case

**Comment by travankor** [comment]
What about https://github.com/zenorogue/hyperrogue/blob/master/Makefile.simple?

**Comment by xelxebar** [comment]
> What about https://github.com/zenorogue/hyperrogue/blob/master/Makefile.simple?

That makefile contains no `install` target.

**Comment by xtraeme** [comment]
cannot be built with SDL2?

**Comment by xelxebar** [comment]
> cannot be built with SDL2?

Not without a non-trivial patch:

    $ grep -r 'SDL/SDL.h' masterdir/builddir/hyperrogue-11.3f
    masterdir/builddir/hyperrogue-11.3f/.travis.yml:    (cd /usr/local/include && ln -sf SDL/SDL.h)
    masterdir/builddir/hyperrogue-11.3f/configure.ac:AC_CHECK_HEADERS([SDL/SDL.h], [], AC_MSG_ERROR([SDL/SDL.h header was not found]))
    masterdir/builddir/hyperrogue-11.3f/fake-mobile.cpp:#include <SDL/SDL.h>
    masterdir/builddir/hyperrogue-11.3f/savepng.cpp:#include <SDL/SDL.h>
    masterdir/builddir/hyperrogue-11.3f/sysconfig.h:#include <SDL/SDL.h>

    $ xlocate SDL/SDL.h
    SDL-devel-1.2.15_12	/usr/include/SDL/SDL.h

Apparently, SDL2 isn't just a drop-in replacement for SDL. For the former, some headers look to be named like `SDL_*` while others are `SDL2_*`. Same goes for function names.

**Comment by xelxebar** [comment]
Replaced the symlink hack with a `CXXFLAGS` setting. Much better.

**Comment by Johnnynator** [comment]
You do miss some licenses `CC BY-SA 4.0, CC BY 3.0, Public Domain` for various sounds. Also set `nocross="Reason"` instead of setting archs

**Comment by xelxebar** [comment]
> You do miss some licenses `CC BY-SA 4.0, CC BY 3.0, Public Domain` for various sounds.
Very nice catch. I found the same by scrounging around and ending up at `sounds/credits.txt`. Did you do the same? That was very non-obvious to me.

What are some general thinks to keep in mind to avoid missing licenses in the future? Games and anything with artwork seems likely to have several licenses for those parts.

> Also set `nocross="Reason"` instead of setting archs.
Much better. Thank you.

> Also is it possible to use system DejaVuSans-Bold instead of installing a own one?
I have no idea. Submit a patch if you like.

**Comment by Johnnynator** [comment]
> What are some general thinks to keep in mind to avoid missing licenses in the future? Games and anything with artwork seems likely to have several licenses for those parts.

Ideally a program/game would just list everything in the root directory, so one wouldn't need to dig. Assets (Sounds, Graphics) are usually licensed differently, so it is usually unlikely that everything is GPL, digging into the subdirs and looking for textfiles there usually helps find these.

> I have no idea. Submit a patch if you like.

Just checked the code, it is currently hardcoded. And would require a patch for hyperrogue itself. So just keep it, makes it the 12th copy of it in the repo :D

**Comment by Johnnynator** [comment]
I missed the soundtrack license. (`./hyperrogue-music.txt`)
```
HyperRogue soundtrack under the Creative Commons BY-SA 3.0 license, http://creativecommons.org/licenses/by-sa/3.0/
```

**Comment by xelxebar** [comment]
Thanks for the tips. Just for kicks, I went ahead and did a grep carpet bomb:

    $ grep -ri license masterdir/builddir/hyperrogue-11.3f

which turned up lots of different results. It looks like `mingw.*` stuff is under the BSD 2-clause while `dll/*` is under the MIT. Those are clearly Windows-specific, but how should these be handled?

**Comment by Johnnynator** [comment]
>  Those are clearly Windows-specific, but how should these be handled?

Everything that doesn't get installed or used in the build process is (mostly) irrelevant, and doesn't need to be mentioned in the license field.

**Comment by xelxebar** [comment]
Beautiful. Thanks.

**Pull request closed without merging**